### ISSUE TITLE
Collector Select - Default values & remembering selections

### ISSUE THREAD

**Issue by mattttt** [raintank/grafana#144]
When using the collector select radio button/dropdowns, selections for each section are not being remembered. So if you select a group of tags, switch to Static (just to window shop, hit update), and then go back to Dynamic, you've lost your previous selection:

![collector-select](https://cloud.githubusercontent.com/assets/2886187/7841202/f94358da-0468-11e5-87d1-146b9c31fad6.gif)

With that in mind, we should have default selections for both tag and collectors, so there is never an empty state.

**Comment by mattttt** [comment]
Had a good discussion with @woodsaj on this. Not as open/shut as I originally thought.

I proposed this behavior:

![collector-behavior](https://cloud.githubusercontent.com/assets/2886187/7915041/eb13ff6a-0848-11e5-9345-fa645a33881f.gif)

But his main concerns being that the UI suggests otherwise, and that it's a bit dangerous to treat it that way. My main goal is to provide forgiveness to the user if they dont know what they want, and end up switching back and forth a couple times before ultimately deciding. 

Will have to give it some additional thought.

**Comment by mattttt** [comment]
Additional thoughts from @woodsaj, documented here so I can re-read when Im working on it:

"if the user has 'static' with all as the initial state.  Then selects 'dynamic', and picks a few tags.  then without clicking 'update' selects 'static' again, then selects 'dynamic' again and it shows their previously picked tags they may assume that their choice has already been saved. from my perspective, if a system remembers something, i asume that is saved."

**Comment by mattttt** [comment]
Perhaps even changing the CTA in the dropdown from 'Update' to 'Confirm' would be a good first step. It's a small change, but at least it feels more firm and decisive, compared to Update.

**Comment by nopzor1200** [comment]
Interesting discussion. 

I just want to throw in 2c here: I feel the biggest problem is that in the rest of Grafana there's very few update buttons, settings are automatically saved when you "click out". 

I'm not super opinionated about what is "right" and what is "wrong", but I do feel that it should be consistent across the board and that inconsistency will cause a lot of confusion.

Changing the word from Update to Confirm might be a good thing, they seem practically synonymous in this situation to me personally.

**Comment by mattttt** [comment]
Great feedback. Couple things in no particular order:

(1) Update Settings is the main CTA in each section, so while the words themselves may be similar, it's important to not mix action words in the same areas. For example, if you just hit "Update" in the dropdown, do you assume it's

(2) I like the idea of consistency across the interface as well. I'm struggling here, because while it's the beginnings of convention within grafana (albeit a bit different than this form), I'm not convinced it's the right convention to roll out across the system at large. 

This is an imaginary scenario that I envision - curious if you guys could see this happening:

- Nathan creates a new endpoint for eyeball.domain.com, and configures it. 
- A month later, his co-worker Joe goes to the config for eyeball.domain.com to see what collectors are monitoring the Ping check. 
- He sees that it's being monitored from a Static Footprint, but he wonders if he could switch it to tags. 
- He clicks Dynamic Footprint, clicks on a couple tags to see what collectors would be used - but ultimately deciding that he doesn't want to make any changes before talking to Nathan. 
- Clicks off the dropdown, gets a message that says "Endpoint Updated"
- *Moment of panic* - "Wait, updated? What collectors did Nathan have selected? I didnt want to switch it. I didnt click Update."

On the dashboard, conversely, I believe the auto-updating has less consequences because it's just impacting the current view (until the dashboard is saved - which is an explicit action). 

To add some forgiveness to the interface but maintain this convention the alternative could be to add an "Undo" into the Updated messages. This could get a little tricky if there were multiple things happening in succession, but it's certainly worth thinking about. 

tl;dr: Definitely worth continuing to think about.

**Comment by woodsaj** [comment]
I do like raj's idea.  The select box should not be treated any differently to other select boxes.

For example.

in the HTTP check, there is a select box for 'Method'.  When this is changed the user does not need to confirm the change.  However we also don't immediately post the change to the backend until the user clicks the 'update HTTP settings' button.

The collector selection should be the same. I think it is a more consistent approach If we remove the 'confirm' and 'cancel' buttons from the select menu and just have the choice acknowledged but not posted to the backend when the user clicks off the select popover,  The user can always abandon the change by navigating away from the page without clicking the 'update HTTP settings' button

**Comment by woodsaj** [comment]
@mattttt is this good to close now?

**Comment by mattttt** [comment]
This good to me now.
